File: src/clients/python/examples/load_html.py
# /// script
# requires-python = ">=3.13"
# dependencies = [
#     "justbe-webview",
# ]
#
# [tool.uv.sources]
# justbe-webview = { path = "../" }
# ///
import asyncio

from justbe_webview import (
    WebView,
    Options,
    ContentHtml,
    Notification,
)


async def main():
    print("Creating webview")
    config = Options(
        title="Load Html Example",
        load=ContentHtml(
            html="<h1>Initial html</h1>",
            # Note: This origin is used with a custom protocol so it doesn't match
            # https://example.com. This doesn't need to be set, but can be useful if
            # you want to control resources that are scoped to a specific origin like
            # local storage or indexeddb.
            origin="example.com",
        ),
        devtools=True,
    )

    async with WebView(config) as webview:

        async def handle_start(event: Notification):
            await webview.open_devtools()
            await webview.load_html("<h1>Updated html!</h1>")

        webview.on("started", handle_start)

    print("Webview closed")


if __name__ == "__main__":
    asyncio.run(main())

File: src/clients/go/examples/load_html.go
package main

import (
	"context"
	"fmt"
	"log"

	"github.com/justbe-engineering/webview-client-go/webview"
)

func main() {
	ctx := context.Background()

	loadContent := webview.ContentFrom(webview.NewHtmlContent("<h1>Initial html</h1>", "example.com"))
	
	options := webview.Options{
		Title:    "Load Html Example",
		Devtools: boolPtr(true),
		Load: &loadContent,
	}

	wv, err := webview.NewWebView(ctx, options)
	if err != nil {
		log.Fatal(err)
	}
	defer wv.Close()

	// Register event handler for "started" event
	wv.On("started", func(event interface{}) {
		fmt.Println("WebView started!")

		// Open dev tools
		if err := wv.OpenDevTools(); err != nil {
			log.Printf("Failed to open dev tools: %v", err)
		}

		// Load new HTML content
		if err := wv.LoadHTML("<h1>Updated html!</h1>", ""); err != nil {
			log.Printf("Failed to load HTML: %v", err)
		}
	})

	// Register event handler for window close
	wv.On("closed", func(event interface{}) {
		fmt.Println("WebView closed!")
	})

	// Wait for the webview to close
	if err := wv.Wait(); err != nil {
		log.Printf("WebView exited with error: %v", err)
	}
}

// Helper functions for pointer types
func boolPtr(b bool) *bool {
	return &b
}

func strPtr(s string) *string {
	return &s
}